### DISCUSSION TITLE
Failed to find guild, or unknown type for channel 1033xxxx 15

### DISCUSSION THREAD

**Discussion by HRK44** [discordjs/discord.js#8792] [Q&A]
### Which package is this bug report for?

discord.js

### Issue description

I noticed this issue with my code, it's trying to pull a channel or a guild with id 1033xxx but there's no guild with that id in my system and my code only pulls guild info and members info, so not sure why it's showing that id in my logs.

### Code sample

All my discord js calls:

```typescript                     

                            const communityGuild = await this.client.guilds.fetch(community.guild.id);`


                            community.guild.name = communityGuild.name;
                            community.guild.memberCount = communityGuild.memberCount;
                            community.guild.bannerURL = communityGuild.bannerURL({ extension: 'png' });
                            community.guild.iconURL = communityGuild.iconURL({ extension: 'png' });

                            const xxx= communityGuild.roles.cache.find(r => (r.name === 'xxx'));

                            if (!communityGuild.members.me?.permissions.toArray().includes('ManageRoles')) { ... }
```

+ add user roles `discordMember.roles.add(roleToCheck)`

### Package version

14.3.0

### Node.js version

v16.13.1

### Priority this issue should have

Low (slightly annoying)

### Which partials do you have configured?

Not applicable (subpackage bug)

### Which gateway intents are you subscribing to?

Guilds

**Comment by almostSouji** [comment]
Where is that ID coming from? Persistent storage? Is it a string (it needs to be) or a number at any point? If the latter, Discord ids are too big to fit into JS numbers, so they are truncated and will be invalid as ids (no exact id, no querying).

If that's not the concern this needs a stack trace and debug logs (for example the guild ID coming in).

**Reply by HRK44** [reply]
This is coming from the discord debug (`client.on('debug')`), the ID is a discord id of I'm guessing a channel?

```
 
00:58:42.387
Emitting close and reconnecting again.
 
00:58:42.387
[1] 2022-10-28T05:58:42.011Z debug: [WS => Shard 0] [CLOSE]
 
00:58:44.434
[1] 2022-10-28T05:58:44.207Z debug: [WS => Shard 0] Failed to connect to the gateway, requeueing...
 
00:58:44.434
[1] 2022-10-28T05:58:44.207Z debug: [WS => Manager] Shard Queue Size: 1; continuing in 5 seconds...
 
00:58:44.434
[1] 2022-10-28T05:58:44.119Z debug: [WS => Shard 0] Clearing the HELLO timeout.
 
00:58:44.434
[1] 2022-10-28T05:58:44.207Z debug: [WS => Shard 0] [CLOSE]
 
00:58:44.434
    Event Code: 1006
 
00:58:44.434
    Clean     : false
 
00:58:44.434
    Reason    :
 
00:58:47.519
[1] 2022-10-28T05:58:47.058Z debug: [WS => Shard 0] An open connection was found, attempting an immediate identify.
 
00:58:48.547
[1] 2022-10-28T05:58:47.059Z debug: [WS => Shard 0] [IDENTIFY] Shard 0/1 with intents: 98303
 
00:59:01.797
    at Timeout._onTimeout (/app/node_modules/discord.js/src/managers/GuildMemberManager.js:536:16)
 
00:59:01.797
    at listOnTimeout (node:internal/timers:559:17)
 
00:59:01.797
    at processTimers (node:internal/timers:502:7)
 
00:59:01.797
[1] 2022-10-28T05:59:01.686Z warn: Error [GuildMembersTimeout]: Members didn't arrive in time.
```

Some more logs above ^

My bot is running fine, then this happens randomly. Connection seems to close, then this error message appears with an id that's not matching any guild id in my system (and my code doesnt pull/interact with channels) `Failed to find guild, or unknown type for channel 1033xxxx 15`

**Reply by almostSouji** [reply]
so, have you given my initial guess a thought?
i didn't mean the ws debug log, that's pretty much useless if you expect that to come from a rest call.
you could inspect `community.guild.id` and see if it is what you expect it to be, as suggested.

you also haven't addressed if this is coming from a database system or similar, which datatype it is saved as, and if it may be truncated due to js max number limits, because it arrives as number in js.
